canopy_phot_mod.F90

```diff
@@ -0,0 +1,53 @@
+module canopy_phot_mod
+
+    implicit none
+
+contains
+
+!:::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
+    SUBROUTINE CANOPY_PHOT( HCM, ZK, FAFRACK, LAI, CLU, COSZEN, RJCORR )
```

Comment:
just making them `elemental` might be sufficient to be able to call with the full profile

Reply:
@zmoon I agree on all accounts, and I have eliminated the need for HCM and ZK and pass in full profiles as the calculation naturally accounts for above and below canopy.  You can see these changes as well as the change in RJCORR variable name, and explicit information on Eq. and page number for the photolysis attenuation calculation in the next [commit](https://github.com/noaa-oar-arl/canopy-app/pull/20/commits/580eca3f68f58426c9ba66dc36a0fa8a66c4bf5b), and can resolve the conversation if satisfied.

However, seems like the canopy_wind calculation will need to be called separately somewhere anyway because of multiple conditions in the submodule that are not only controlled by FAFRACK profile, e.g.,:

       zkhcm = ZK/HCM
        z0g = Z0GHCM*HCM
        hol = HCM/MOL
       ! Nondimensional canopy wind speed term that dominates near the ground:
        if (ZK >= z0g .and. ZK <= HCM) then
            canbot = log(zkhcm/Z0GHCM)/log(1.0_rk/Z0GHCM)
        else if (ZK >= 0 .and. ZK <= z0g) then
            canbot = 0.0  !No-slip condition at surface (u=0 at z=0)
        else
            canbot = 1  ! should be zk > hcm
        end if
...
...
...
       if (ZK <= HCM) then       !at or below canopy top --> modify uc winds
            CANWIND = uc*canbot*cantop
        else
            CANWIND = UBZREF       !above canopy top constant and equal to reference winds
        end if

Reply:
Well I didn't try it, but I suspect it would work automatically for profile (and still for point) if you made it an `elemental subroutine`. But if you would prefer just putting loop inside the subroutine, that's fine, I do think that is preferable to doing it outside.

Reply:
@zmoon OK.  I tried it and this seems like we should make this an idea for its own issue outside of this PR.  For canopy_wind,  I would have to change more of the code when making it an elemental subroutine and passing in whole array, because of the following errors:

canopy_wind_mod.F90:90:54:

   90 |             write(*,*) "critical problem: hcan <= zpd"
      |                                                      1
Error: IO UNIT in WRITE statement at (1) must be an internal file in a PURE procedure
canopy_wind_mod.F90:137:97:

  137 |                 write(*,*) 'wrong namelist option', RSL_OPT, 'only 0 (MOST) or 1 (RSL) available'
      |                                                                                                 1
Error: IO UNIT in WRITE statement at (1) must be an internal file in a PURE procedure
canopy_wind_mod.F90:149:86:

  149 |             write(*,*)  'Uc cannot be <= 0 at  ',uc , ' in canopy_wind calc...exiting'
      |                                                                                      1
Error: IO UNIT in WRITE statement at (1) must be an internal file in a PURE procedure
canopy_wind_mod.F90:91:24:

   91 |             call exit(2)
      |                        1
Error: Subroutine call to intrinsic ‘exit’ at (1) is not PURE
canopy_wind_mod.F90:138:28:

  138 |                 call exit(2)
      |                            1
Error: Subroutine call to intrinsic ‘exit’ at (1) is not PURE

Reply:
Ah yes, `elemental` implies `pure` and printing not allowed in `pure`. So I guess putting the loop inside is the way to go in order to keep those messages. Separate PR fine for sure.